src/com/amazon/corretto/crypto/provider/HpkeGen.java

```diff
@@ -0,0 +1,49 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.corretto.crypto.provider;
+
+import java.security.*;
+import java.security.spec.AlgorithmParameterSpec;
+
+public class HpkeGen extends KeyPairGeneratorSpi {
+  private final AmazonCorrettoCryptoProvider provider_;
+  private HpkeParameterSpec spec = null;
+
+  HpkeGen(AmazonCorrettoCryptoProvider provider) {
+    Loader.checkNativeLibraryAvailability();
+    provider_ = provider;
+  }
+
+  /** Generates a new HPKE key and returns a pointer to it. */
+  private static native long generateEvpHpkeKemKeyFromSpec(int hpke_kem_id);
+
+  @Override
+  public void initialize(final AlgorithmParameterSpec params, final SecureRandom rnd)
+      throws InvalidAlgorithmParameterException {
+    if (params instanceof HpkeParameterSpec) {
+      // TODO: do validation
+      spec = (HpkeParameterSpec) params;
+    } else {
+      throw new InvalidAlgorithmParameterException("Unsupported AlgorithmParameterSpec: " + spec);
+    }
+  }
+
+  @Override
+  public void initialize(final int keysize, final SecureRandom rnd)
+      throws InvalidParameterException {
+    throw new InvalidParameterException(
+        "Cannot initialize a KEM key with keysize, must use AlgorithmParameterSpec.");
+  }
+
+  @Override
+  public KeyPair generateKeyPair() {
+    if (spec == null) {
+      // TODO: support default spec?
```

Comment:
Interesting question. On the one hand, that would make this easier to use. On the other hand, there are consequences to getting the default wrong and having to live with it. I'd lean towards not setting a default unless we find a good reason to support it.

Reply:
That's fair, will update.